doc/source/tutorial/stats/rv_infrastructure.md

```diff
@@ -416,53 +434,82 @@ The NLLF of the parameters estimated using `fit` are lower:
+
+```{code-cell} ipython3
+# params = stats.weibull_min.fit(data, loc=0)  # careful! this provides loc=0 as a *guess*
+params = stats.weibull_min.fit(data, floc=0)
+params
+```
+
+```{code-cell} ipython3
+stats.weibull_min.nnlf(params, data)
+```
+
+By relying on a `fit` method to "just work", users are encouraged to neglect some of these important details. 
+
+Instead, we suggest that it is almost as simple to fit distributions to data using the generic optimization tools provided by SciPy. However, this allows the user to perform the fit according to their preferred objective function, provides options when things don't work as expected, and helps the user to avoid the most common pitfalls.
+
+For instance, suppose our desired objective function is indeed the negative log likelihood function. It is simple to define as a function of the shape parameter `c` and scale only, leaving the location at its default of zero.
 
 ```{code-cell} ipython3
 def nllf(params):
-    c, loc, scale = params
-    X = Weibull(c=c) * scale + loc
+    c, scale = params
+    X = Weibull(c=c) * scale
     return -X.logpdf(data).sum()
 
-nllf(params)
+c_, loc_, scale_ = params
+nllf((c_, scale_))
 ```
 
 To perform the minimization, we can use `scipy.optimize.minimize`.
 
 ```{code-cell} ipython3
 from scipy import optimize
 eps = 1e-10  # numerical tolerance to avoid invalid parameters
-lb = [eps, -10, eps]  # lower bound on `c`, `location`, and `scale`
-ub = [10, np.min(data)-eps, 10]  # upper bound on `c`, `location`, and `scale`
-x0 = [1, 1, 1]  # guess to get optimization started
+lb = [eps, eps]  # lower bound on `c` and `scale`
+ub = [10, 10]  # upper bound on `c` and `scale`
+x0 = [1, 1]  # guess to get optimization started
 bounds = optimize.Bounds(lb, ub)
 res = optimize.minimize(nllf, x0, bounds=bounds)
 res
 ```
 
-Although the value of `fun` (the NLLF) is not quite as low as with the solution provided by `weibull_min.fit`, the PDFs are not so different.
+The value of the objective function is almost identical, and the PDFs are indistinguishable.
 
 ```{code-cell} ipython3
 import matplotlib.pyplot as plt
-x = np.linspace(2, 10, 300)
+x = np.linspace(0, 15, 300)
 
-c, loc, scale = res.x
-X = Weibull(c=c)*scale + loc
+c, scale = res.x
+X = Weibull(c=c)*scale
 plt.plot(x, X.pdf(x), '-', label='numerical optimization')
 
 c, loc, scale = params
-Y = Weibull(c=c)*scale + loc
+Y = Weibull(c=c)*scale
 plt.plot(x, Y.pdf(x), '--', label='`scipy.stats.fit`')
 
-plt.hist(data, bins=np.linspace(2, 10, 30), density=True, alpha=0.1)
+plt.hist(data, bins=np.linspace(0, 20, 30), density=True, alpha=0.1)
 plt.xlabel('x')
 plt.ylabel('pdf(x)')
 plt.legend()
 ```
 
-## New Features
+However, wi
```

Comment:
Hah I got distracted. But this is where I was going to show that we can easily adjust the objective function.